app/demo-loop/demo-loop.cc

```diff
@@ -113,30 +148,35 @@ void run(std::istream* is, OutputManager* output)
 
     // Save results to ROOT MC truth output file when possible
     std::shared_ptr<RootFileManager> root_manager;
-    std::shared_ptr<StepCollector> step_collector;
-    std::shared_ptr<RootStepWriter> step_writer;
-
-    if (!run_args.mctruth_filename.empty())
     {
-        CELER_LOG(info) << "Writing ROOT MC truth output at "
-                        << run_args.mctruth_filename;
+        std::shared_ptr<StepCollector> step_collector;
+        std::shared_ptr<RootStepWriter> step_writer;
```

Comment:
That was more for readability on my side while updating the code, and I left it as it seemed easier to visualize where the ROOT-related calls are happening. If you prefer, I can just remove the extra `{ }`

Reply:
OK, it made the diff a little weird but I think it's fine for the function itself. However now I'm questioning whether it should be in the `demo-loop.cc` or in the `LDemoIO` setup? It should probably be in its own function at this point, though, since the `run` function is getting so long.

Reply:
I've made a separate helper function for it, cleaning up `run()`. It is still in the `demo-loop` because I'm still pondering if we should expand LDemoIO even more by moving both the filtering function (`root_step_writer_filter()`) and `init_root_mctruth_output()`...